### ISSUE TITLE
Crash when opening suggestions window, lookup window, dictionary editor

### ISSUE THREAD

**Issue by Teteros** [openstenoproject/plover#1085]
## Summary
Plover terminates when more than one dictionary editor window is opened at a time.

```
2019-08-22 09:00:01,003 [Dummy-1] INFO: loading dictionary: /home/tete/.local/share/plover/user.json
[New Thread 0x7fffd0db8700 (LWP 1307)]
2019-08-22 09:00:01,004 [Dummy-1] INFO: loading dictionary: /home/tete/.local/share/plover/commands.json
[New Thread 0x7fffc7fff700 (LWP 1308)]
2019-08-22 09:00:01,004 [Dummy-1] INFO: loading dictionary: /home/tete/.local/share/plover/main.json
[New Thread 0x7fffc77fe700 (LWP 1309)]
[Thread 0x7fffd0db8700 (LWP 1307) exited]
[Thread 0x7fffc7fff700 (LWP 1308) exited]
[Thread 0x7fffc77fe700 (LWP 1309) exited]
2019-08-22 09:00:01,546 [Dummy-1] INFO: loaded 3 dictionaries in 0.542s
AttributeError: __getattribute__

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/lib/python3.7/site-packages/plover/gui_qt/dictionaries_widget.py", line 323, in on_activate_cell
    self._edit([self._config_dictionaries[row]])
  File "/usr/lib/python3.7/site-packages/plover/gui_qt/dictionaries_widget.py", line 319, in _edit
    editor = DictionaryEditor(self._engine, [d.path for d in dictionaries])
  File "/usr/lib/python3.7/site-packages/plover/gui_qt/dictionary_editor.py", line 280, in __init__
    self.setupUi(self)
  File "/usr/lib/python3.7/site-packages/plover/gui_qt/dictionary_editor_ui.py", line 94, in setupUi
    QtCore.QMetaObject.connectSlotsByName(DictionaryEditor)
SystemError: <built-in function connectSlotsByName> returned a result with an error set

Thread 1 "python" received signal SIGABRT, Aborted.
0x00007ffff7df5755 in raise () from /usr/lib/libc.so.6
(gdb) bt
#0  0x00007ffff7df5755 in raise () from /usr/lib/libc.so.6
#1  0x00007ffff7de0851 in abort () from /usr/lib/libc.so.6
#2  0x00007ffff54948b6 in QMessageLogger::fatal(char const*, ...) const () from /usr/lib/libQt5Core.so.5
#3  0x00007ffff5a08048 in pyqt5_err_print() [clone .cold] () from /usr/lib/python3.7/site-packages/PyQt5/QtCore.so
#4  0x00007ffff5b700d6 in PyQtSlotProxy::unislot(void**) () from /usr/lib/python3.7/site-packages/PyQt5/QtCore.so
#5  0x00007ffff5b70b60 in PyQtSlotProxy::qt_metacall(QMetaObject::Call, int, void**) () from /usr/lib/python3.7/site-packages/PyQt5/QtCore.so
#6  0x00007ffff56c8acd in QMetaObject::activate(QObject*, int, int, void**) () from /usr/lib/libQt5Core.so.5
#7  0x00007ffff27b7ae0 in QTableWidget::cellActivated(int, int) () from /usr/lib/libQt5Widgets.so.5
#8  0x00007ffff56c8a11 in QMetaObject::activate(QObject*, int, int, void**) () from /usr/lib/libQt5Core.so.5
#9  0x00007ffff2752746 in QAbstractItemView::activated(QModelIndex const&) () from /usr/lib/libQt5Widgets.so.5
#10 0x00007ffff2761063 in QAbstractItemView::mouseDoubleClickEvent(QMouseEvent*) () from /usr/lib/libQt5Widgets.so.5
#11 0x00007ffff2da7414 in sipQTableWidget::mouseDoubleClickEvent(QMouseEvent*) () from /usr/lib/python3.7/site-packages/PyQt5/QtWidgets.so
#12 0x00007ffff2528d5b in QWidget::event(QEvent*) () from /usr/lib/libQt5Widgets.so.5
#13 0x00007ffff25d9b1f in QFrame::event(QEvent*) () from /usr/lib/libQt5Widgets.so.5
#14 0x00007ffff2760a40 in QAbstractItemView::viewportEvent(QEvent*) () from /usr/lib/libQt5Widgets.so.5
#15 0x00007ffff2da89dc in sipQTableWidget::viewportEvent(QEvent*) () from /usr/lib/python3.7/site-packages/PyQt5/QtWidgets.so
#16 0x00007ffff569c6d3 in QCoreApplicationPrivate::sendThroughObjectEventFilters(QObject*, QEvent*) () from /usr/lib/libQt5Core.so.5
#17 0x00007ffff24e54c4 in QApplicationPrivate::notify_helper(QObject*, QEvent*) () from /usr/lib/libQt5Widgets.so.5
#18 0x00007ffff24eec6c in QApplication::notify(QObject*, QEvent*) () from /usr/lib/libQt5Widgets.so.5
#19 0x00007ffff2bc2bbf in sipQApplication::notify(QObject*, QEvent*) () from /usr/lib/python3.7/site-packages/PyQt5/QtWidgets.so
#20 0x00007ffff569c9c2 in QCoreApplication::notifyInternal2(QObject*, QEvent*) () from /usr/lib/libQt5Core.so.5
[...]
```
## Reproducing
Run plover and start two instances of the dictionary editor (double click on a dictionary twice)

## Plover Version
current git master (Plover 4.0.0.dev8+161.g[f4e4536](f4e45364fc5f58745c459f878d13ec6442497f46))

## System
Arch Linux x64, QT 5.13.0

**Comment by nimble0** [comment]
Also happens when opening the lookup window. Downgrading python to 3.7.3 fixes it. There's more information about it here:

https://www.riverbankcomputing.com/pipermail/pyqt/2019-August/042026.html

**Comment by user202729** [comment]
As mentioned in the link, the code itself is also buggy. In this case it's because `setupUi` tries to access `text_font` property which relies on an unassigned attribute of `self`. Fix:

```
--- a/plover/gui_qt/suggestions_widget.py
+++ b/plover/gui_qt/suggestions_widget.py
@@ -22,10 +22,10 @@ class SuggestionsWidget(QWidget, Ui_SuggestionsWidget):
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
-        self.setupUi(self)
         self._translation_char_format = QTextCharFormat()
         self._strokes_char_format = QTextCharFormat()
         self._strokes_char_format.font().setStyleHint(QFont.Monospace)
+        self.setupUi(self)
 
     def append(self, suggestion_list):
         scrollbar = self.suggestions.verticalScrollBar()
```

**Comment by morinted** [comment]
Would updating PyQT address this issue or will we need to modify our tools to not except?

**Comment by nimble0** [comment]
I'm not getting the error now with Python 3.8.1 and PyQt5 5.11.2

I can't reproduce the error at all now though, (I think) because I can't install any versions of PyQt5 below 5.11 on Python 3.8 (some sip requirement error).

**Comment by morinted** [comment]
Still getting this on python 3.7.6 with

```
altgraph                  0.17
appdirs                   1.4.3
appnope                   0.1.0
attrs                     19.3.0
Babel                     2.7.0
biplist                   1.0.3
bleach                    3.1.4
certifi                   2018.10.15
cffi                      1.14.0
chardet                   3.0.4
check-manifest            0.41
cmarkgfm                  0.4.2
Cython                    0.29.16
dmgbuild                  1.3.3
docutils                  0.16
ds-store                  1.1.2
hidapi                    0.7.99.post21
idna                      2.9
importlib-metadata        1.6.0
jsonschema                3.2.0
mac-alias                 2.0.7
macholib                  1.14
more-itertools            8.2.0
packaging                 20.3
pep517                    0.8.2
pip                       20.0.2
pkginfo                   1.5.0.1
plover                    4.0.0.dev8
plover-dictionary-builder 0.1.0
plover-emoji              0.0.3
plover-last-translation   0.0.2
plover-layout-display     0.1.0
plover-plugins-manager    0.5.14
plover-python-dictionary  0.5.10
plover-retro-case         0.1.2
plover-stenograph-usb     0.0.4
plover-stitching          0.1.0
plover-treal              1.0.1
plover-vlc-commands       0.6.5
pluggy                    0.13.1
py                        1.8.1
pycparser                 2.20
Pygments                  2.6.1
pyobjc-core               5.1.1
pyobjc-framework-Cocoa    5.1.1
pyobjc-framework-Quartz   5.1.1
pyparsing                 2.4.6
PyQt5                     5.11.3
PyQt5-sip                 4.19.13
pyrsistent                0.16.0
pyserial                  3.4
pytest                    5.4.1
pytz                      2019.3
pyusb                     1.0.2
readme-renderer           25.0
requests                  2.23.0
requests-cache            0.5.2
requests-futures          1.0.0
requests-toolbelt         0.9.1
setuptools                46.1.3
setuptools-scm            3.5.0
simplefuzzyset            0.0.12
six                       1.11.0
spectra-lexer             10.0.0
toml                      0.10.0
tqdm                      4.45.0
twine                     1.15.0
urllib3                   1.25.8
wcwidth                   0.1.7
webencodings              0.5.1
wheel                     0.34.2
zipp                      3.1.0
```

**Comment by morinted** [comment]
Upgraded to PyQt5-sip==4.19.14 and PyQt5==5.12.3 and it persists.

**Comment by morinted** [comment]
Tried with our official version of python on Mac (3.6.8) and it's working fine. If it's fixed in 3.8 as well, then I think we can close this issue and just request that you change your python version.

All the best.
